### ISSUE TITLE
Member page metadata retrieval errors

### ISSUE THREAD

**Issue by amenconi** [soonaverse/app#138]
**Bug Description:**
When the member page loads (any page with the member profile drawer open such as member activity, awards, spaces etc.) in some cases the member or spaces array returns malformed data (i.e. in the array of spaces you might find [object object] string instead of space meta data such as the UID and other properties).  This will cause the Data Service to make invalid calls to the API and the HTML template will attempt to access UID property on objects where they don't exist.

**Reproduce:**
Navigate to "https://soonaverse.com/member/0x3ffbc221b2215d6e7d9c863fb44616a1ae042017/badges" (one of my Soonaverse accounts) with dev tools console open and you will see some failed calls to API as well as number errors such as:
```
ERROR TypeError: Cannot read properties of undefined (reading 'uid')
    at DefaultIterableDiffer.trackByUid [as _trackByFn] (member-about.component.ts:67:17)
    at DefaultIterableDiffer.check (core.mjs:26476:36)
    at DefaultIterableDiffer.diff (core.mjs:26457:18)
    at NgForOf.ngDoCheck (common.mjs:3092:42)
    at callHook (core.mjs:2498:18)
    at callHooks (core.mjs:2457:17)
    at executeCheckHooks (core.mjs:2389:5)
    at selectIndexInternal (core.mjs:9147:17)
    at Module.ɵɵadvance (core.mjs:9136:5)
    at MemberAboutComponent_Template (member-about.component.html:152:10)
```

**Expected behavior:**
Loading the page shouldn't produce any of these errors regardless of account loaded.

**Screenshots**
Image: 
![image](https://github.com/soonaverse/app/assets/7841580/cbd7996b-4a2f-44ab-aa1c-1f3463dd9878)
Video: https://github.com/soonaverse/app/assets/7841580/54145a1c-2be2-4420-9446-3139c8633d57

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Browser: Chrome
 - Version: 122.0.6261.112

**Comment by emmap3-do** [comment]
@amenconi there is an old issue I found in production last year, related to the uid, that might be fixed with these changes.
I'm adding it here for reference: https://github.com/soonaverse/app/issues/53

**Comment by amenconi** [comment]
@emmap3-do ah yes, familiar with this function.  The checkout button pops up from the header component.  Had to refactor much of that code to make this feature work with the cart and bulk buy purchase processes as well.

Hopefully this PR or the shopping cart fixes issue #53, if that bug remains after merging both I can revisit and review specifically to fix but I'm hopeful it'll be resolved with one of these.

Thanks.

**Comment by emmap3-do** [comment]
I just tested and, unfortunately, bug is still there for issue #53
